docs/calculating-effective-entry-date.md

```diff
@@ -0,0 +1,163 @@
+
+## 2. Same-Day ACH Transaction Limit
+
+* The per-entry limit for same-day ACH is **$1,000,000**
+* Entries exceeding this amount are not eligible for same-day settlement
+* In a batch containing entries over the limit, those entries will have their Effective Entry Date adjusted to the next banking day and remaining items will still be eligible for same-day settlement
+
+## 3. Credit Entries
+
+For credit entries, the Effective Entry Date must be either:
+* The **same banking day** as the processing date for **Same Day Entries**, or
+* **One or two banking days** following the processing date for other entries
+
+## 4. Debit Entries
+
+For debit entries, the Effective Entry Date must be either:
+* The **same banking day** as the processing date for **Same Day Entries**, or
+* **One banking day** following the processing date for other entries
+
+**Note:** Receiving banks may place holds on ACH debits per their funds availability policies, separate from these settlement times.
+
+## 5. Effective Entry Date - Edge Cases
+
+### ENR Entries
+* For Automated Enrollment Entries (ENR), the Effective Entry Date field must be **space-filled**
+* ENR entries cannot be Same Day Entries
+
+### Notifications of Change
+* The ACH Operator does **not edit** the Effective Entry Date field for Notifications of Change
+
+### Return Entries
+* The ACH Operator may verify that the Effective Entry Date is properly formatted and may replace the field's existing content with the current processing date if the Effective Entry Date is invalid
+
+# Calculating Effective Entry Dates Using moov-io/base Functions
+
+## Overview
+
+The moov-io/base package provides functionality for calculating valid Effective Entry Dates when creating ACH batch headers, ensuring compliance with NACHA rules and Federal Reserve processing schedules.
+
+## Creating Batch Headers
+
+The NewBatchHeader function creates a BatchHeader object that requires several key fields to be populated according to NACHA rules:
+
+```go
+bh := NewBatchHeader()
+bh.ServiceClassCode = "220"                  // 220 for credits only
+bh.StandardEntryClassCode = PPD              // PPD, CCD, WEB, etc.
+bh.CompanyName = "ACME Corporation"          // Your company name
+bh.CompanyIdentification = "121042882"       // Your company ID
+bh.CompanyEntryDescription = "PAYROLL"       // Transaction description
+bh.ODFIIdentification = "12104288"           // Your bank's routing number
+
+bh.EffectiveEntryDate = "240102"             // YYMMDD format
+```
+
+## Banking Day Functions
+
+The moov-io/base Time package provides these essential functions for determining valid banking days:
+
+* `AddBankingTime(hours, minutes, seconds int)`: Increments time by specified duration while respecting banking hours (9am-5pm ET)
+* `AddBankingDay(d int)`: Adds the specified number of valid banking days to a date
+* `IsBankingDay()`: Validates if a date is a banking day by checking both holidays and weekends
```

Comment:
Can we add links to all of these functions?

e.g. https://pkg.go.dev/github.com/moov-io/base#Time.IsBankingDay

Reply:
Added links to each of the base functions.